CExcercises/Ex3-genericAdder.c

```diff
@@ -57,15 +57,17 @@ int main (int argc, char **argv)
 
 int genericPlus(const void* array, size_t elementSize, size_t fieldOffset, size_t amountOfElements)
 {
-	int i = 0;
+	// Like te return tyoe is int, assume that all the fields to add are int
+	int indexOfStructs = 0;
 	int sumResult = 0;
-	struct S * sArray  = (struct S *) array; // Do with pointer
+	struct S * sArray  = (struct S *) array; // Cast to Struct S
 
-	for(i=0; i < amountOfElements; i = i + 1)
+	for(indexOfStructs=0; indexOfStructs < amountOfElements; indexOfStructs ++)
 	{
-		int algo = (int)fieldOffset;
-		int algo2 = (int) sArray;
-		//sumResult = sumResult + *((int *) 
+		int * tempPoniter = (int *) ((int)sArray + indexOfStructs * (elementSize) + fieldOffset);
+		// A int cast of the struct is needed to the processor doesn't add the length of the struct
+		// example: sArray + x = sArray + x * (sizeof(struct S))  It's characteristic of C compiler
```

Comment:
I know that you are adding comments to remember some things, but usually is better to have self documented code that over explained code